test/integration/integration_test.h

```diff
@@ -4,24 +4,25 @@
 
 #include "gtest/gtest.h"
 
-class IntegrationTest : public BaseIntegrationTest, public testing::Test {
+class IntegrationTest : public BaseIntegrationTest,
+                        public testing::TestWithParam<Network::Address::IpVersion> {
 public:
   /**
-   * Global initializer for all integration tests.
+   * Initializer for individual integration tests.
    */
-  static void SetUpTestCase() {
+  void SetUp() override {
```

Comment:
Got it. Yes I agree removing statics will make for better tests so if you are up for it, please go for it.